core/src/main/java/org/apache/calcite/sql/SqlNumericLiteral.java

```diff
@@ -90,7 +90,7 @@ public boolean isExact() {
   @Override public String toValue() {
     BigDecimal bd = getValueNonNull();
     if (isExact) {
-      return getValueNonNull().toString();
+      return getValueNonNull().toPlainString();
```

Comment:
Why not use `bd.toPlainString()`?

And the `bd` should be `final`.

Reply:
Good point, I pushed these changes. Since the PR is so small I have already squashed the two commits.